torch/utils/data/datapipes/iter/routeddecoder.py

```diff
@@ -1,47 +1,49 @@
-from torch.utils.data import IterDataPipe
+from io import BufferedIOBase
+from typing import Any, Callable, Iterable, Iterator, Sized, Tuple
+
+from torch.utils.data import IterDataPipe, functional_datapipe
 from torch.utils.data.datapipes.utils.decoder import (
     Decoder,
     basichandlers as decoder_basichandlers,
-    imagehandler as decoder_imagehandler)
+    imagehandler as decoder_imagehandler,
+    _default_key_fn)
 
-from typing import Iterable, Iterator, Union, List, Tuple, Any, Callable
-from io import BufferedIOBase
 
+@functional_datapipe('decode')
 class RoutedDecoderIterDataPipe(IterDataPipe[Tuple[str, Any]]):
     r""" :class:`RoutedDecoderIterDataPipe`.
 
-    Iterable datapipe to decode binary streams from input iterables,
-    yield pathname and decoded binary stream in a tuple.
+    Iterable datapipe to decode binary streams from input DataPipe, yield pathname
+    and decoded data in a tuple.
     args:
         datapipe: Iterable datapipe that provides pathname and binary stream in tuples
-        handlers: user defined decoder handlers, if None, basic and image decoder handlers will be set as default
-        length: a nominal length of the datapipe
+        handlers: Optional user defined decoder handlers. If None, basic and image decoder
+            handlers will be set as default. If multiple handles are provided, the priority
+            order follows the order of handlers (the first handler has the top priority)
+        key_fn: Function for decoder to extract key from pathname to dispatch handlers.
+            Default is set to extract file extension from pathname
```

Comment:
Maybe we need to add a comment saying that if this returns anything other than the extension, the default handlers will not work and need to be also overwritten.

Reply:
Make sense to me. Will add several notes to this DataPipe.